common/persistence/cassandraPersistence.go

```diff
@@ -455,7 +455,8 @@ const (
 )
 
 var (
-	defaultVisibilityTimestamp = time.Time{} // The zero value of type Time is January 1, year 1, 00:00:00.000000000 UTC
+	defaultDateTime            = time.Date(2000, time.January, 1, 0, 0, 0, 0, time.UTC)
+	defaultVisibilityTimestamp = common.UnixNanoToCQLTimestamp(defaultDateTime.UnixNano())
```

Comment:
(1) What's the purpose of this field ? CreateShard() / GetShard() all use the default placeholder today ?
(2) I don't think the problem here is gocql client expecting a BigInt instead of time.Time. I believe you need to pass in *time.Time instead of time.Time because it expects an interface{} type. So, you should be able to leave the previous const as such, just add & before.
(3) Can you add unit test for Shard CRUD ? (Ok, if this comes in a followup diff)